### PR TITLE
Add tuple-new and tuple-new-n primitives and test cases

### PR DESCRIPTION
**Pull request by kjmcqueen**
https://rchain.atlassian.net/browse/ROS-256

### CODE REVIEW

File: roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala [hunk 1]
```diff
@@ -226,6 +226,67 @@ object tuple {
...
+  }
+
+  /**
+    * Define the tuple-new-n primitive.
+    * This returns a new Tuple containing the specified n duplicated Obs
+    *
+    * See warning in tplNew about the rationale for ignoring ARG(0).
+    * 	(tplNewN dummy n init)
+    *
+    * e.g.(tplNewN [] 3 'a) ===>	['a 'a 'a]
```
Comment (tymm) [inline roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala]: This seems to be an error in the original comments.
I guess we should fix it to: `(tuple-new-n [] 3 'a)`.

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala [hunk 2]
```diff
@@ -239,4 +239,56 @@ class TupleSpec extends FlatSpec with Matchers {
     tplTail.fn(newCtxt) should be(Right(Tuple(empty)))
   }
 
+  /** tuple-new */
+  "tplNew" should "correctly create a new Tuple with n Obs" in {
+    val tup = Seq(Fixnum(1), Fixnum(2), Fixnum(3), Fixnum(4), Fixnum(5), Fixnum(6))
+
+    val newCtxt =
+      ctxt.copy(
+        nargs = 8,
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Should be `7`, right?

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala [hunk 3]
```diff
@@ -239,4 +239,56 @@ class TupleSpec extends FlatSpec with Matchers {
...
+    val newCtxt =
+      ctxt.copy(
+        nargs = 3,
+        argvec = Tuple(Seq(Fixnum(0), Fixnum(6), Fixnum(1)))
+      )
+    tplNewN.fn(newCtxt) should be(Right(tup))
+  }
+
+  it should "return NIL for n<=0" in {
+    val empty = Tuple(Seq.empty)
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Where do you use `empty`?

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala [hunk 4]
```diff
@@ -239,4 +239,56 @@ class TupleSpec extends FlatSpec with Matchers {
...
+    val newCtxt =
+      ctxt.copy(nargs = 1, argvec = Tuple(Tuple(empty)))
+
+    tplNew.fn(newCtxt) should be(Right(Tuple(empty)))
+  }
+
+  /** tuple-new-n */
+  "tplNewN" should "correctly create a new Tuple with n duplicate Obs" in {
+    val tup   = Tuple(6, Fixnum(1))
+    val empty = Seq.empty
```
Comment (tymm) [inline roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Where do you use `empty`?

File: roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala (hunk 1 above)
Comment (kjmcqueen) [reply roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala]: I agree.  Done.

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala (hunk 2 above)
Comment (kjmcqueen) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: I think so. Done.

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala (hunk 3 above)
Comment (kjmcqueen) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Left over.  Done.

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala (hunk 4 above)
Comment (kjmcqueen) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Another left over.  Done.

File: roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala (hunk 2 above)
Comment (kjmcqueen) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: Nope.  It's because of that ignored first parameter.

Comment (tymm) [reply roscala/src/test/scala/coop/rchain/rosette/prim/TupleSpec.scala]: You still have to count in the ignored first element though.

**Pull request merged**